packages/protocolbeat/src/discovery-ui/multi-view/store.ts

```diff
@@ -0,0 +1,220 @@
+import { z } from 'zod'
+import { create } from 'zustand'
+
+export const PANEL_IDS = ['list', 'values', 'nodes', 'code', 'preview'] as const
+export type PanelId = (typeof PANEL_IDS)[number]
+
+export type Panel = { id: PanelId; size: number }
+
+export type State = {
+  mouse: { x: number; y: number }
+  panels: Panel[]
+  layouts: Panel[][]
+  selectedLayout: number
+  active: PanelId | undefined
+  fullScreen: PanelId | undefined
+  hover: PanelId | undefined
```

Comment:
It's confusing that hover is changed by a function called `pickUp`. I think this variable can be renamed to `pickedUp` for example. Anything that indicates that the PanelId in this variable is changed by the `pickUp` function.